src/crafting_gui.cpp

```diff
@@ -408,9 +408,17 @@ const recipe *select_crafting_recipe( int &batch_size )
 
             ypos = 0;
 
+            auto qry = trim( filterstring );
+            std::string qry_comps( "" );
+            bool is_prefixed = qry.size() > 2 && qry[1] == ':';
+            if( is_prefixed ) {
+                if( qry[0] == 'c' )
```

Comment:
Styling. Every `if` must have braces.
Also, you can merge those two `if`s.

Reply:
And than you don't even need the boolean variable. Or you can just use `std::string::compare` to do the comparison:
```C++
if( qry.compare( 0, 2, "c:" ) == 0 ) {
```

Also: you don't need to explicitly initialize a string with an empty string literal. `std::string qry_cmps;` is enough.